pallene/checker.lua

```diff
@@ -455,7 +455,7 @@ function Checker:check_stat(stat)
             elseif loop_type._tag == "types.T.Float" then
                 stat.step = ast.Exp.Float(stat.limit.loc, 1.0)
             else
-                error("impossible")
+                typedecl.tag_error(loop_type._tag, "incorrect loop step type.")
```

Comment:
Nice. This is exactly the kind of situation that benefits the most from a custom error message.

Small suggestion: what do you think about changing the error message to `loop type is not a number type`?

Reply:
Yeah, this error message is a bit unclear. fixed.